src/main/cocoa/CocoaCairoSurface.mm

```diff
@@ -56,10 +57,13 @@ static inline cairo_antialias_t decode_antialiasing(const Font &f)
                 return CAIRO_ANTIALIAS_DEFAULT;
             }
 
-            CocoaCairoSurface::CocoaCairoSurface(CocoaDisplay *dpy, size_t width, size_t height):
+            CocoaCairoSurface::CocoaCairoSurface(CocoaDisplay *dpy, NSWindow *window, size_t width, size_t height):
                 ISurface(width, height, ST_IMAGE)
             {
                 pDisplay        = dpy;
+                pCocoaWindow    = window;
+                fWidth          = width;
```

Comment:
We have `nWidth` and `nHeight` in parent class `ws::ISurface` with protected access.